parser/msg_parser.c

```diff
@@ -925,6 +925,9 @@ int parse_msg(char* buf, unsigned int len, struct sip_msg* msg)
 	return 0;
 
 error:
+	/* Free the SIP message if it has already been created to prevent memory leaks */
```

Comment:
tiny improvement, more on clarity than in performance. As only `parse_headers` may induce a potential leak (line 875 here), I would do there a `goto error2` and here, on top of the existing `error` label I would add the new `error2` doing the `free_sip_msg()` and continuing with the `error` sequence.

Reply:
Thank you for the review. I fixed in updated commit.